### ISSUE TITLE
Segmentation fault on exit using Ubuntu and Python 3.9

### ISSUE THREAD

**Issue by dstein64** [pytorch/pytorch#50858]
## 🐛 Bug

When using Python 3.9 on Ubuntu, a segmentation fault arises when exiting the Python interpreter after loading a model and making a call to `backward()` (the model loaded and the call to `backward` are independent in the example below). This occurs using the latest stable release (1.7.1) and the nightly release (installed 2021/1/20) of PyTorch. This does not occur when using Python 3.8 or earlier. This does not occur on macOS or Windows.

The problem reported here is a minimal reproducible example, derived from the same problem that I encountered in an actual context using PyTorch.

## To Reproduce

On Ubuntu, create a file, `main.py`, with the following code:

```python
import torch
import torchvision.models as models

print('start')
resnet18 = models.resnet18(pretrained=True)
loss = torch.tensor([1.0], requires_grad=True)
loss.backward()
print('end')
```

Run the code with Python 3.9.

```
$ python3.9 main.py
```

This will output:

```
start
end
Segmentation fault
```

The problem does not occur when either the `resnet18` assignment line or the `loss.backward()` line is commented out.

Occasionally there is a hang instead of a segmentation fault.

## Expected behavior

No segmentation fault.

```
start
end
```

## Environment

 - PyTorch Version (e.g., 1.0): 1.7.1+cpu
 - OS (e.g., Linux): Ubuntu 20.04.1
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): N/A
 - Python version: Python 3.9.0+
 - CUDA/cuDNN version: N/A
 - GPU models and configuration: N/A
 - Any other relevant information: The same problem also occurs with Ubuntu 18.04 running Python 3.9.1, as shown [here](https://github.com/dstein64/pytorch-50858/actions/runs/500472568) (Python 3.8 does not have the problem).

## Additional context

Possibly related Issues:
- https://github.com/pytorch/pytorch/issues/47161
- https://github.com/huggingface/transformers/issues/9191

**Comment by ngimel** [comment]
Possibly duplicate of #50014

**Comment by ngimel** [comment]
cc @malfet

**Comment by dstein64** [comment]
Thanks @ngimel. I've posted a comment in #50014, including any information from above that was not already in that ticket, and will close this ticket as a duplicate.
